src/app/content/practiceQuestions/components/ShowPracticeQuestions.tsx

```diff
@@ -96,30 +107,33 @@ const ShowPracticeQuestions = () => {
       data-testid='show-practice-questions-body'
       data-analytics-region='PQ popup'
     >
-      {section ? <SectionTitle dangerouslySetInnerHTML={{ __html: section.title }} /> : null}
-      <QuestionsWrapper>
-        <QuestionsHeader>
-          <FormattedMessage id='i18n:practice-questions:popup:questions'>
-            {(msg: string) => msg}
-          </FormattedMessage>
-        </QuestionsHeader>
-        <ProgressBar total={questionsCount} activeIndex={currentQuestionIndex} />
+      <Filters />
+      <ShowPracitceQuestionsContent>
+        {section ? <SectionTitle dangerouslySetInnerHTML={{ __html: section.title }} /> : null}
+        <QuestionsWrapper>
+          <QuestionsHeader>
+            <FormattedMessage id='i18n:practice-questions:popup:questions'>
+              {(msg: string) => msg}
+            </FormattedMessage>
+          </QuestionsHeader>
+          <ProgressBar total={questionsCount} activeIndex={currentQuestionIndex} />
+          {
+            selectedSectionHasPracticeQuestions && currentQuestionIndex === null
+              ? <IntroScreen />
+              : null
+          }
+        </QuestionsWrapper>
         {
-          selectedSectionHasPracticeQuestions && currentQuestionIndex === null
-            ? <IntroScreen />
+          section && linkToTheSection
+            ? <StyledContentLink href={linkToTheSection} target='_blank' data-analytics-label='Read' >
+              <FormattedMessage id='i18n:practice-questions:popup:read'>
```

Comment:
whitespace is wrong. also, this component is becoming overloaded. there is too much data being selected and too many conditions in the rendering. you could make a separate component for this link that selects `linkToTheSection` and internally returns `null` when required.

`IntroScreen` probably should stay the way you have it because we'll have to swap it out for the actual questions so it won't be self contained logic

`section` i imagine will always be set, when could it not be?

Reply:
We are setting the `section` when we are opening a modal and we are resting it when we close it / on `locationChange` so if you open the modal short after navigating to the other page it will still be a `null`.

It is done this way because we don't want to remember user choices when he closed and then reopened the modal. But maybe we don't have to clear this data and then we can set selected section on `locationChange`?

Reply:
perhaps it could be organized in such a way that the section is set after clicking the button but before the modal is shown? that way we know we can display the filters immediately

Reply:
I've moved this component to `<LinkToSection section={section} />`

and I've moved `setSelectedSection` action to the `practiceQuestionButton` component and removed `practiceQuestionOpenHook`